### PR TITLE
Fix thread list being ordered based on all updates

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/25522

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix thread list being ordered based on all updates ([\#3458](https://github.com/matrix-org/matrix-js-sdk/pull/3458)). Fixes vector-im/element-web#25522.

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -1957,7 +1957,7 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
             }
         }
 
-        this.on(ThreadEvent.Update, this.onThreadUpdate);
+        this.on(ThreadEvent.NewReply, this.onThreadReply);
```
Comment (richvdh) [inline src/models/room.ts:1960]: Is this a fix for something?

Comment (t3chguy) [reply src/models/room.ts:1960]: Yes sorry PR took title of the first commit (again), fixed it now, the issue it fixes is in the PR description first line

**Review by richvdh — APPROVED** [review]
seems fine

File: spec/integ/matrix-client-event-timeline.spec.ts [hunk 2]
```diff
@@ -1568,7 +1568,8 @@ describe("MatrixClient event timelines", function () {
                 // Test adding a second event to the first thread
                 const thread = room.getThread(THREAD_ROOT.event_id!)!;
                 thread.initialEventsFetched = true;
-                const prom = emitPromise(room, ThreadEvent.Update);
+                const prom = emitPromise(room, ThreadEvent.NewReply);
+                respondToEvent(THREAD_ROOT_UPDATED);
                 respondToEvent(THREAD_ROOT_UPDATED);
                 respondToEvent(THREAD_ROOT_UPDATED);
                 respondToEvent(THREAD_ROOT_UPDATED);
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply spec/integ/matrix-client-event-timeline.spec.ts:1575]: Great question, one being targeted by https://github.com/matrix-org/matrix-js-sdk/pull/3450

**Pull request merged**